pal/src/thread/thread.cpp

```diff
@@ -2820,3 +2820,32 @@ void GetCurrentThreadStackLimits(ULONG_PTR* lowLimit, ULONG_PTR* highLimit)
     *highLimit = (ULONG_PTR) stackbase;
 #endif
 }
+
+static thread_local ULONG_PTR s_cachedThreadHighLimit = 0;
```

Comment:
nit pick: `s_cachedThreadHighLimit` variable name doesn't indicate it's stack related.